src/pray.c

```diff
@@ -1006,7 +1006,14 @@ gcrownu()
         break;
     }
     case A_NONE:
-        /* nothing to do */
+        /* OK, we don't get an artifact, but surely Moloch
+         * can at least offer His own blessing? */
+        if (ok_wep(obj) && !obj->oartifact && !(obj->oprops & ITEM_FIRE))
```

Comment:
Oops, just realized the player can wield stackable weapons.  50 daggers of fire is probably too much!

Reply:
And fixed.  Hopefully my force-push hadn't exploded anyone's PC.